### PR TITLE
CICD Tests

### LINKED ISSUE #14: Run unit tests

### PR DESCRIPTION
**Pull request by markspec**
Update CICD pipeline to:
* build tests
* run tests

If tests run this will resolve #14  

This can only be tested by creating a PR

### CODE REVIEW

File: .github/workflows/cmake_build.yaml [hunk 1]
```diff
@@ -18,3 +18,7 @@ jobs:
         uses: threeal/cmake-action@main
         with:
           cxx-compiler: ${{ matrix.compiler }}
+      - name: Build tests
+        run: cd build && make mdio_acceptance_test  && make mdio_variable_test && make mdio_dataset_test
+      - name: Run tests
+        run: cd build/mdio/  && ./mdio_acceptance_test && ./mdio_variable_test && ./mdio_dataset_test
```
Comment (BrianMichell) [inline .github/workflows/cmake_build.yaml:24]:
```suggestion
      - name: Build tests
        run: cd build && make -j
      - name: Run tests
        run: cd build/mdio/  && ./mdio_acceptance_test && ./mdio_variable_test && ./mdio_dataset_test && ./mdio_dataset_factory_test && ./mdio_dataset_validator_test && ./mdio_stats_test && ./mdio_utils_test && ./mdio_variable_collection_test
```
There are a handful of optional cloud-store tests as well, but those shouldn't be part of the CICD pipelines.

**Review by BrianMichell — COMMENTED** [review]
The acceptance test currently runs our limited Python compatibility regression testing. We will need to make the zarr and xarray Python modules available before the tests will pass.

**Comment by markspec** [comment]
@BrianMichell , please can you fix the failing tests/

**Review by BrianMichell — APPROVED** [review]
LGTM

**Pull request merged**